cmd/cloudFoundryDeleteService.go

```diff
@@ -1,45 +1,106 @@
 package cmd
 
 import (
+	"bytes"
+	"fmt"
+	"strings"
+
 	"github.com/SAP/jenkins-library/pkg/command"
 	"github.com/SAP/jenkins-library/pkg/log"
 	"github.com/SAP/jenkins-library/pkg/telemetry"
 )
 
-func cloudFoundryDeleteService(CloudFoundryDeleteServiceOptions cloudFoundryDeleteServiceOptions, telemetryData *telemetry.CustomData) error {
+func cloudFoundryDeleteService(options cloudFoundryDeleteServiceOptions, telemetryData *telemetry.CustomData) {
 
 	c := command.Command{}
 
 	// reroute command output to logging framework
 	c.Stdout(log.Entry().Writer())
 	c.Stderr(log.Entry().Writer())
 
-	cloudFoundryLogin(CloudFoundryDeleteServiceOptions, &c)
+	var err error
 
-	err := cloudFoundryDeleteServiceFunction(CloudFoundryDeleteServiceOptions.CfServiceInstance, &c)
-	if err != nil {
-		cloudFoundryLogout(&c)
+	err = cloudFoundryLogin(options, &c)
+
+	if err == nil && options.CfDeleteServiceKeys == true {
+		err = cloudFoundryDeleteServiceKeys(options, &c)
+	}
+
+	if err == nil {
+		err = cloudFoundryDeleteServiceFunction(options.CfServiceInstance, &c)
+	}
+
+	var logoutErr error
+
+	if err == nil {
+		logoutErr = cloudFoundryLogout(&c)
+		if logoutErr != nil {
+			log.Entry().
+				WithError(logoutErr).
+				Fatal("Error while logging out occured.")
+		}
+	} else if err != nil {
+		logoutErr = cloudFoundryLogout(&c)
+		if logoutErr != nil {
+			log.Entry().
+				WithError(logoutErr).
+				Fatal("Error while logging out occured.")
+		}
 		log.Entry().
 			WithError(err).
-			Fatal("Failed to delete Service!")
-		return err
+			Fatal("Error occured.")
+	}
+}
+
+func cloudFoundryDeleteServiceKeys(options cloudFoundryDeleteServiceOptions, c execRunner) error {
+
+	log.Entry().Info("Deleting inherent Service Keys")
+
+	var cfFindServiceKeysScript = []string{"service-keys", options.CfServiceInstance}
+
+	var serviceKeyBytes bytes.Buffer
+	c.Stdout(&serviceKeyBytes)
+
+	err := c.RunExecutable("cf", cfFindServiceKeysScript...)
+
+	if err != nil {
+		return fmt.Errorf("Failed to Delete Service Key, most likely your service doesn't exist: %w", err)
 	}
 
-	cloudFoundryLogout(&c)
+	if len(serviceKeyBytes.String()) == 0 {
+		log.Entry().Info("No service key could be retrieved for your requested Service")
+		return err
```

Comment:
Please make it clear that there is no error by returning `nil` here instead of `err`.

Reply:
Ok changed it for the whole code to return `nil` instead of `err` where no errow is thrown.